subsys/settings/src/settings.c

```diff
@@ -32,27 +31,113 @@ void settings_init(void)
 
 int settings_register(struct settings_handler *handler)
 {
-	if (settings_handler_lookup(handler->name)) {
-		return -EEXIST;
+	int rc;
+	struct settings_handler *ch;
+
+	k_mutex_lock(&settings_lock, K_FOREVER);
+
+	SYS_SLIST_FOR_EACH_CONTAINER(&settings_handlers, ch, node) {
```

Comment:
OK, but it should be unit-tested.

Reply:
^^ Not relevant anymore: comment was to handler sorting which is not implemented in final,